### PR TITLE
feat(domains) Add test to ensure that future routes work with customer-domains

### PR DESCRIPTION
**Pull request by markstory**
Going forward we need to define both org slug based routes and routes for customer domains. Currently that requirement is being held up with code review. This change adds some automation to ensure that for each path based route we also have a domain based route defined.

I figured having checks for the routes being defined is enough but I can go further and attempt to resolve the components and verify that those are the same too if folks feel that level of check is relevant.

Refs HC-668

### CODE REVIEW

File: static/app/routes.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+function extractRoutes(rootRoute: any): Record<string, RouteComponent> {
+  const routeTree = createRoutes(rootRoute);
+  const routeMap: Record<string, RouteComponent> = {};
+
+  // A queue of routes we need to visit
+  const visitQueue: RouteMetadata[] = [{leadingPath: '', route: routeTree[0]}];
+  while (visitQueue.length > 0) {
+    const current = visitQueue.pop();
+    if (!current) {
```
Comment (dashed) [inline static/app/routes.spec.tsx:43]: Was this to overcome typescript's limitations? We already know that `visitQueue.length > 0` by line 43.

File: static/app/routes.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+      if (!domainRoutes[domainPath]) {
+        mismatch.push({slug: path, domain: domainPath});
+      }
+      if (mismatch.length > 0) {
+        const routelist = mismatch
+          .map(item => `- slug: ${item.slug}\n  domain: ${item.domain}`)
+          .join('\n');
+        throw new Error(
+          `Could not matching URLs for the following ${mismatch.length} routes:\n\n` +
```
Comment (dashed) [inline static/app/routes.spec.tsx:111]:
```suggestion
          `Unable to find matching URLs for the following ${mismatch.length} routes:\n\n` +
```

**Review by dashed — APPROVED** [review]
Looks good to me.

File: static/app/routes.spec.tsx (hunk 1 above)
Comment (markstory) [reply static/app/routes.spec.tsx:43]: Yes typescript was complaining about the possibility of undefined.

**Review by dashed — APPROVED** [review]

**Pull request merged**